lib/src/async_cache.dart

```diff
@@ -62,8 +62,11 @@ class AsyncCache<T> {
     }
     if (_cachedValueFuture == null) {
       _cachedValueFuture = callback();
-      await _cachedValueFuture;
-      _startStaleTimer();
+      try {
+        await _cachedValueFuture;
+      } finally {
+        _startStaleTimer();
+      }
     }
     return _cachedValueFuture!;
```

Comment:
The code here looks slightly problematic. It returns a future *after* awaiting that same future, and reads and null-checks `_cachedValueFuture` repeatedly.

I'd write everything from line 63 as:

```dart
  var result = _cachedValueFuture ??= callback();
  try {
    return await result;
  } finally {
    _startStaleTimer();
  }
```

This should not change the behavior of the code, but it also doesn't need to repeatedly read `_cachedValueFuture`, and it doesn't have to await it twice (once in the `await`, and once in the `return`).

Reply:
Thanks, it's much better this way.